src/rules/no_undef.rs

```diff
@@ -0,0 +1,345 @@
+
+    assert_lint_ok::<NoUndef>("var a; function f() { a = 1; }");
+
+    assert_lint_ok::<NoUndef>("Object; isNaN();");
+  }
+
+  #[test]
+  fn ok_3() {
+    assert_lint_ok::<NoUndef>("toString()");
+
+    assert_lint_ok::<NoUndef>("hasOwnProperty()");
+
+    assert_lint_ok::<NoUndef>("function evilEval(stuffToEval) { var ultimateAnswer; ultimateAnswer = 42; eval(stuffToEval); }");
+  }
+
+  #[test]
+  fn ok_4() {
+    assert_lint_ok::<NoUndef>("typeof a");
+
+    assert_lint_ok::<NoUndef>("typeof (a)");
+
+    assert_lint_ok::<NoUndef>("var b = typeof a");
+  }
+
+  #[test]
+  fn ok_5() {
+    assert_lint_ok::<NoUndef>("typeof a === 'undefined'");
+
+    assert_lint_ok::<NoUndef>("if (typeof a === 'undefined') {}");
+
+    assert_lint_ok::<NoUndef>(
+      "function foo() { var [a, b=4] = [1, 2]; return {a, b}; }",
+    );
+  }
+
+  #[test]
+  fn ok_6() {
+    assert_lint_ok::<NoUndef>("var toString = 1;");
+
+    assert_lint_ok::<NoUndef>("function myFunc(...foo) {  return foo;}");
+
+    // TODO(kdy1): Parse as jsx
+    // assert_lint_ok::<NoUndef>(
+    //   "var React, App, a=1; React.render(<App attr={a} />);",
+    // );
+  }
+
+  #[test]
+  fn ok_7() {
+    assert_lint_ok::<NoUndef>(
+      "var console; [1,2,3].forEach(obj => {\n  console.log(obj);\n});",
+    );
+
+    assert_lint_ok::<NoUndef>(
+      "var Foo; class Bar extends Foo { constructor() { super();  }}",
+    );
+
+    assert_lint_ok::<NoUndef>(
+      "import Warning from '../lib/warning'; var warn = new Warning('text');",
+    );
+  }
+
+  #[test]
+  fn ok_8() {
+    assert_lint_ok::<NoUndef>("import * as Warning from '../lib/warning'; var warn = new Warning('text');");
+
+    assert_lint_ok::<NoUndef>("var a; [a] = [0];");
+
+    assert_lint_ok::<NoUndef>("var a; ({a} = {});");
+  }
+
+  #[test]
+  fn ok_9() {
+    assert_lint_ok::<NoUndef>("var a; ({b: a} = {});");
+
+    assert_lint_ok::<NoUndef>("var obj; [obj.a, obj.b] = [0, 1];");
+
+    assert_lint_ok::<NoUndef>(
+      "(foo, bar) => { foo ||= WeakRef; bar ??= FinalizationRegistry; }",
+    );
+  }
+
+  #[test]
+  fn ok_10() {
+    assert_lint_ok::<NoUndef>("Array = 1;");
+
+    assert_lint_ok::<NoUndef>("class A { constructor() { new.target; } }");
+
+    assert_lint_ok::<NoUndef>(r#"export * as ns from "source""#);
+  }
+
+  #[test]
+  fn ok_11() {
+    assert_lint_ok::<NoUndef>("import.meta");
+  }
+
+  #[test]
+  fn err_1() {
+    assert_lint_err::<NoUndef>("a = 1;", 0);
+
+    assert_lint_err::<NoUndef>("var a = b;", 8);
+
+    assert_lint_err::<NoUndef>("function f() { b; }", 15);
+  }
+
+  #[test]
+  fn err_2() {
+    assert_lint_err::<NoUndef>("window;", 0);
+
+    assert_lint_err::<NoUndef>("require(\"a\");", 0);
+
+    // TODO(kdy1): parse as jsx
+    // assert_lint_err::<NoUndef>("var React; React.render(<img attr={a} />);", 0);
+  }
+
+  #[test]
+  fn err_3() {
+    // TODO(kdy1): parse as jsx
+    // assert_lint_err::<NoUndef>("var React, App; React.render(<App attr={a} />);", 0);
```

Comment:
Test case should be parseable as `tsx` already

Reply:
How can I parse it as tsx?

Reply:
Sorry, I was mistaken... You'd need to add additional helper to pass syntax in tests - by default it uses `TypeScript` syntax. 

https://github.com/denoland/deno_lint/blob/313ef64e548ec971bb6b5ab279aa363315c8269b/src/test_util.rs#L8-L14

If it won't break other tests, feel free to change default syntax to TSX.